share/completions/gsettings.fish

```diff
@@ -0,0 +1,98 @@
+# Fish completions for glibs's "gsettings" configuration tool
+
+function __fish_complete_gsettings_args
+    set -l schema_commands get monitor writable range describe set reset reset-recursively list-keys list-children list-recursively
+    set -l key_commands get monitor writable range describe set reset
+
+    set -l schemadir
+    set -l cmd (commandline -poc)
+    set -e cmd[1]
+
+    if set -q cmd[2]; and string match -q -- '--schemadir' $cmd[1]
+        # use eval to ensure the expansion of ~
+        set schemadir --schemadir (eval echo $cmd[2])
```

Comment:
After further consideration, just leave the expansion bit out for now. This is lacking across the board, and should not be fixed in a single completion script. It's also more than just tilde, it's also variables and braces.